freud/box.pyx

```diff
@@ -921,6 +942,43 @@ cdef class Box:
                             "positional argument: L")
         return cls(Lx=L, Ly=L, Lz=0, xy=0, xz=0, yz=0, is2D=True)
 
+    @classmethod
+    def from_box_lengths_and_angles(
+        cls, L1, L2, L3, alpha, beta, gamma, dimensions=None,
+    ):
+        r"""Construct a box from lengths and angles.
+
+        All the angles provided must be between 0 and :math:`\pi`.
+
+        Args:
+            L1 (float):
+                The length of the first lattice vector.
+            L2 (float):
+                The length of the second lattice vector.
+            L3 (float):
+                The length of the third lattice vector.
+            alpha (float):
+                The angle between second and third lattice vector in radians.
+            beta (float):
+                The angle between first and third lattice vector in radians.
+            gamma (float):
+                The angle between the first and second lattice vector in radians.
+            dimensions (int):
+                The number of dimensions (Default value = :code:`None`).
+
+        Returns:
+            :class:`freud.box.Box`: The resulting box object.
+        """
+        a1 = np.array([L1, 0, 0])
+        a2 = np.array([L2 * np.cos(gamma), L2 * np.sin(gamma), 0])
+        a3x = np.cos(beta)
+        a3y = (np.cos(alpha) - np.cos(beta) * np.cos(gamma)) / np.sin(gamma)
+        a3z = np.sqrt(1 - a3x**2 - a3y**2)
```

Comment:
This line appears to take the square root of a negative number at times, resulting in NaNs.

Reply:
some combinations of angles are not possible-there is likely a constraint on their sum or something like that. Is this taken into account?